lib/utils/build_context_ext.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+extension Theming on BuildContext {
+  TextTheme get textTheme => Theme.of(this).textTheme;
+}
+
+extension Localization on BuildContext {
```

Comment:
We can merge into one, like below: 
```suggestion
extension BuildContextExtension on BuildContext {
TextTheme get textTheme => Theme.of(this).textTheme;
AppLocalizations get localization => AppLocalizations.of(this)!;
}
```